### PR TITLE
[WIP] Stop overriding default assembly load context

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
cc @jaredpar @tmat

**Comment by tmat** [comment]
@agocke @TyOverby

File: src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs [hunk 1]
```diff
@@ -14,24 +14,17 @@
...
 {
     /// Core CLR compatible wrapper for loading analyzers.
-    internal sealed class CoreClrAnalyzerAssemblyLoader : AssemblyLoadContext, IAnalyzerAssemblyLoader
+    internal sealed class CoreClrAnalyzerAssemblyLoader : IAnalyzerAssemblyLoader
     {
         private readonly Dictionary<string, Assembly> _pathsToAssemblies = new Dictionary<string, Assembly>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, Assembly> _namesToAssemblies = new Dictionary<string, Assembly>();
         private readonly List<string> _dependencyPaths = new List<string>();
         private readonly object _guard = new object();
+        private readonly AssemblyLoadContext _loadContext = AssemblyLoadContext.GetLoadContext(typeof(SomeTypeFromRoslynAssembly).GetTypeInfo().Assembly);
```
Comment (agocke) [inline src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs]: Why don't you just use the current assembly?

Comment (jkotas) [inline src/Compilers/Shared/CoreClrAnalyzerAssemblyLoader.cs]: Because of `Assembly.GetExecutingAssembly()` is not in .NET Core surface

**Comment by agocke** [comment]
LGTM

**Comment by jaredpar** [comment]
test mac please

**Comment by TyOverby** [comment]
LGTM

**Pull request merged**